sdk/src/wallet/account/operations/transaction/high_level/send_amount.rs

```diff
@@ -55,17 +95,68 @@ impl AccountHandle {
         options: Option<TransactionOptions>,
     ) -> crate::wallet::Result<PreparedTransactionData> {
         log::debug!("[TRANSACTION] prepare_send_amount");
-        let mut outputs = Vec::new();
+        let rent_structure = self.client.get_rent_structure().await?;
         let token_supply = self.client.get_token_supply().await?;
 
-        for address_with_amount in addresses_with_amount {
-            let (address, bech32_hrp) = Address::try_from_bech32_with_hrp(address_with_amount.address)?;
+        let account_addresses = self.addresses().await?;
+        let default_return_address = account_addresses.first().ok_or(Error::FailedToGetRemainder)?;
+
+        let local_time = self.client.get_time_checked().await?;
+
+        let mut outputs = Vec::new();
+        for AddressWithAmount {
+            address,
+            amount,
+            return_address,
+            expiration,
+        } in addresses_with_amount
+        {
+            let (address, bech32_hrp) = Address::try_from_bech32_with_hrp(address)?;
+            let return_address = return_address
+                .map(|address| Ok::<_, Error>(Address::try_from_bech32_with_hrp(address)?.0))
+                .transpose()?
+                .unwrap_or(default_return_address.address.inner);
             self.client.bech32_hrp_matches(&bech32_hrp).await?;
-            outputs.push(
-                BasicOutputBuilder::new_with_amount(address_with_amount.amount)?
-                    .add_unlock_condition(AddressUnlockCondition::new(address))
-                    .finish_output(token_supply)?,
-            )
+            // get minimum required amount for such an output, so we don't lock more than required
+            // We have to check it for every output individually, because different address types and amount of
+            // different native tokens require a different storage deposit
+            let storage_deposit_amount = minimum_storage_deposit_basic_native_tokens(
```

Comment:
I think here it should be only for a basic output with address unlock condition, but without SDR(storage deposit return) and expiration unlock conditions
And only in the `} else {` this should be calculated to get the correct amount for this case